File: tests/conformance/fixtures/python/04-gate.py
from sykli import Pipeline

p = Pipeline()

p.task("build").run("make build")

p.gate("approve-deploy").after("build") \
    .gate_strategy("env") \
    .gate_timeout(600) \
    .gate_message("Approve deployment to production?") \
    .gate_env_var("DEPLOY_APPROVED")

p.task("deploy").run("make deploy").after("approve-deploy")

p.emit()

File: tests/conformance/fixtures/elixir/04-gate.exs
use Sykli

pipeline do
  task "build" do
    run "make build"
  end

  gate "approve-deploy" do
    after_ ["build"]
    gate_strategy "env"
    gate_timeout 600
    gate_message "Approve deployment to production?"
    gate_env_var "DEPLOY_APPROVED"
  end

  task "deploy" do
    run "make deploy"
    after_ ["approve-deploy"]
  end
end
|> Sykli.Emitter.validate!()
|> Sykli.Emitter.to_json()
|> IO.puts()
